src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,43 @@
 
 class Animal {
   // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.addAnimal(this);
+  }
+
+  static addAnimal(animal) {
+    this.alive.push(animal);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = this.hidden ? 0 : true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+
+  bite(herbivore) {
+    if (!herbivore.hidden && herbivore.constructor === Herbivore) {
```

Comment:
you have an `instance of` operator
```suggestion
    if (!herbivore.hidden && herbivore instance of Herbivore) {
```